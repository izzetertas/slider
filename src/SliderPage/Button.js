import React from 'react'

const Button = ({ label, onClick, disabled }) => (
  <input
    type='button'
    onClick={onClick}
    value={label}
    disabled={disabled}
    className='button'
  />
)

export default Button;
