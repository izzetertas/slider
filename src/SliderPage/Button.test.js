import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './Button'

describe('<Button />', () => {
  it('should render an input', () => {
    const wrapper = shallow(
      <Button
        value='click me'
        disabled
        onClick={() => console.log('clicked')}
      />)
    expect(wrapper.find('input').length).toBe(1)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should set type correctly', () => {
    const wrapper = shallow(<Button />)
    expect(wrapper.find('input').prop('type')).toEqual('button')
  });

  it('should set label correctly', () => {
    const wrapper = shallow(<Button label='label' />)
    expect(wrapper.find('input').prop('value')).toEqual('label')
  });

  it('should handle onClick correctly', () => {
    const onClickMock = jest.fn()
    const wrapper = shallow(<Button onClick={onClickMock} />)
    wrapper.find('input').simulate('click')
    expect(onClickMock).toHaveBeenCalled()
  });

});