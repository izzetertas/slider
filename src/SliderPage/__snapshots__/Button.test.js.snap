// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render an input 1`] = `
<input
  className="button"
  disabled={true}
  onClick={[Function]}
  type="button"
/>
`;
